packages/gatsby-cli/src/handlers/plugin-babel-utils.ts

```diff
@@ -183,137 +184,181 @@ function buildPluginNode({ name, options, key }): any {
-                  pluginNodes.value.callee.object.elements =
-                    pluginNodes.value.callee.object.elements.map(node => {
-                      const plugin = getPlugin(node)
-
-                      if (plugin.key !== key) {
-                        return node
-                      }
-
-                      if (!plugin.key && plugin.name !== pluginOrThemeName) {
-                        return node
-                      }
-
-                      return buildPluginNode({
-                        name: pluginOrThemeName,
-                        options,
-                        key,
-                      })
-                    })
-                }
-              } else {
-                const plugins = pluginNodes.value.elements.map(getPlugin)
-                const matches = plugins.filter(plugin => {
-                  if (!key) {
-                    return plugin.name === pluginOrThemeName
-                  }
-
-                  return plugin.key === key
-                })
-
-                if (!matches.length) {
-                  const pluginNode = buildPluginNode({
-                    name: pluginOrThemeName,
-                    options,
-                    key,
-                  })
-
-                  pluginNodes.value.elements.push(pluginNode)
-                } else {
-                  pluginNodes.value.elements = pluginNodes.value.elements.map(
-                    node => {
-                      const plugin = getPlugin(node)
-
-                      if (plugin.key !== key) {
-                        return node
-                      }
-
-                      if (!plugin.key && plugin.name !== pluginOrThemeName) {
-                        return node
-                      }
-
-                      return buildPluginNode({
-                        name: pluginOrThemeName,
-                        options,
-                        key,
-                      })
-                    }
-                  )
-                }
-              }
-            } else {
-              if (t.isCallExpression(pluginNodes.value)) {
-                pluginNodes.value.callee.object.elements =
-                  pluginNodes.value.callee.object.elements.filter(node => {
-                    const plugin = getPlugin(node)
-
-                    if (key) {
-                      return plugin.key !== key
-                    }
-
-                    return plugin.name !== pluginOrThemeName
-                  })
-              } else {
-                pluginNodes.value.elements = pluginNodes.value.elements.filter(
-                  node => {
-                    const plugin = getPlugin(node)
-
-                    if (key) {
-                      return plugin.key !== key
-                    }
-
-                    return plugin.name !== pluginOrThemeName
-                  }
-                )
-              }
-            }
-
-            path.stop()
-          },
-        },
+function addPluginsToConfig({
```

Comment:
This is extracted out from existing transform so it can be used for `gatsby-config.js` and `gatsby-config.ts`, nothing should be changed